### ISSUE TITLE
Integrate with Meson

### ISSUE THREAD

**Issue by konradha** [lfortran/lfortran#704]
To integrate LFortran (ie. make it pass all the tests there) with Meson, we can thankfully build on @HaoZeke 's pioneering [work](https://github.com/mesonbuild/meson/pull/9137) -- I started a little on this but realized that we'd need more flags and other things that are standard within Fortran.

**Comment by czgdp1807** [comment]
Are we able to compile SciPy's fortran code base with LFortran?

**Comment by konradha** [comment]
No. That's why I introduced the "SciPy" tag to provide an overview what needs to be done to enable LFortran to compile SciPy.
